java-buildpack/bin/build

```diff
@@ -17,36 +23,65 @@ fi
 
 echo "---> Installing JDK"
 
-if [[ ! -f $launch_dir/jdk.toml ]]; then
-  mkdir -p $launch_dir/jdk
-  jdk_url="https://cdn.azul.com/zulu/bin/zulu8.28.0.1-jdk8.0.163-linux_x64.tar.gz"
-  curl -sfL "$jdk_url" | tar pxz -C $launch_dir/jdk --strip-components=1
+# If it doesn't exist locally, create a JDK cache layer
+# This makes JDK available to subsequent buildpacks as well.
+if [[ -f $cache_dir/jdk.toml ]]; then
+  cached_jdk_url=$(cat "$cache_dir/jdk.toml" | yj -t | jq -r .url 2>/dev/null || echo 'JDK TOML parsing failed')
+fi
+if [[ $jdk_url != $cached_jdk_url ]] ; then
+  rm -rf "$cache_dir"/jdk
+  mkdir -p "$cache_dir"/jdk/env
+  curl -sfL "$jdk_url" | tar pxz -C "$cache_dir"/jdk --strip-components=1
+  echo -e "version = \"$jdk_version\"\nurl = \"$jdk_url\"" > "$cache_dir"/jdk.toml
+
+  echo "$cache_dir"/jdk > "$cache_dir"/jdk/env/JAVA_HOME
+  if [[ -z $LD_LIBRARY_PATH ]]; then
```

Comment:
Could save a bunch of lines with this trick:
https://github.com/buildpack/samples/blob/master/nodejs-buildpack/bin/build#L53

Reply:
My personal opinion is that those just make the buildpack illegible and error prone, I couldn't even make it work if I wanted the ":" on the other side. But feel free to change it,